server.go

```diff
@@ -322,13 +331,63 @@ func (s *ServerImpl) ListDeploys() ([]*Deploy, error) {
 			Id:      fmt.Sprintf("%s-%d", proc.Name, proc.Port),
 			Pid:     proc.Pid,
 			Port:    proc.Port,
+			Set:     proc.Port == setPort,
 			Tracked: false,
 		})
 	}
 	s.checkAllHealth(knownRunningDeploys)
 	return append(knownDeploys, unaccounted...), nil
 }
 
+func (s *ServerImpl) getPortMarkedAsSet() (int, error) {
+
+	url := fmt.Sprintf("http://localhost:%d/;csv", s.endPort)
+	resp, err := http.Get(url)
+	if err != nil {
+	  // Couldn't connect to haproxy don't modify the deploy list
+	  return -1, err
+	}
+	
+	defer resp.Body.Close()
+
+	records, err := csv.NewReader(resp.Body).ReadAll()
+	if err != nil {
+	  return -1, err
+	}
+
+	for i, row := range records {
+	  
+	  // skip header
+	  if i == 0 {
+	    continue
+	  } 
+
+	  // Find the first row that contains a Process name that 
+	  // looks like 'thing-PORT". 
+	  if strings.Contains(row[haProxyPxnameIndex], "thing-") {
```

Comment:
"thing-" is very arbitrary.  So is the way the port number is encoded.  I wouldn't feel comfortable just hard coding it here.

I would pull out a pair of helper functions, e.g. 

```
// ok = true if matching row,  in which case port will be an int with the port
ok, port = parseBackendEntry(row[xyzIndex])

// produces the name of a backend entry
nameBackendEntry(port int) string { ...
```
And have the functions defined next to each other: that way if one gets changed, it's more likely the other will be noticed & updated.

You will need to update the haproxy cfgTemplate to have different vars substituted, i.e. the output of nameBackendEntry instead of the port straight into a string of the form "thing-%APP_PORT%".

Perhaps this whole function belongs more in the haproxy "module" (i.e. .go file).